utils/getweatherdeatails.js

```diff
@@ -0,0 +1,56 @@
+function getWeatherDescription(code) {
+  const weatherDescriptions = {
+    61: "Light rain today, no need to water!",
+    3: "Overcast skies, plants might need extra love.",
+    55: "Heavy drizzle, nature's gentle watering.",
+    56: "Light freezing drizzle, bring plants inside!",
+    57: "Heavy freezing drizzle, your plants need shelter!",
+    65: "Heavy rain, let nature do the watering.",
+    67: "Freezing rain, protect your plants from frostbite!",
+    75: "Heavy snow, your plants want a cozy spot.",
+    77: "Snow grains, tiny ice for your plants!",
+    82: "Heavy rain showers, umbrella for you, rest for your plants.",
+    86: "Heavy snow showers, plants say, 'Nope, not today!'",
+    95: "Thunderstorm, plants prefer to sit this one out.",
+    96: "Thunder with light hail, plants need a safety net!",
+    99: "Thunder with heavy hail, plants are screaming 'HELP!'",
+  };
+
+  return weatherDescriptions[code] || "";
+}
+
+function getAllWeatherDescription(code) {
+  const weatherDescriptions = {
+    0: "Clear sky",
+    1: "Mostly clear",
+    2: "Partly cloudy",
+    3: "Overcast",
+    45: "Fog",
+    48: "Rime fog",
+    51: "Light drizzle",
+    53: "Moderate drizzle",
+    55: "Heavy drizzle",
+    56: "Light freezing drizzle",
+    57: "Heavy freezing drizzle",
+    61: "Light rain",
+    63: "Moderate rain",
+    65: "Heavy rain",
+    66: "Light freezing rain",
+    67: "Heavy freezing rain",
+    71: "Light snowfall",
+    73: "Moderate snowfall",
+    75: "Heavy snowfall",
+    77: "Snow grains",
+    80: "Light rain showers",
+    81: "Moderate rain showers",
+    82: "Heavy rain showers",
+    85: "Light snow showers",
+    86: "Heavy snow showers",
+    95: "Thunderstorm",
+    96: "Thunderstorm with light hail",
+    99: "Thunderstorm with heavy hail",
+  };
+  return weatherDescriptions[code] || "Unknown conditions";
```

Comment:
why do you need to do this twice? Maybe you can give the functions better names so it is very clear what exact purpose they serve like "getWateringInfoByWeatherCode" for the first and "getWeatherDescription" for the second?

Reply:
We differentiate it for two different pages: one is only for the alert icon and the other for the landing page. We don’t want to show everything to the user.